circuit_maintenance_parser/parsers/equinix.py

```diff
@@ -0,0 +1,116 @@
+"""Circuit Maintenance Parser for Equinix Email Notifications."""
+from typing import Any, Dict, List
+import re
+
+from bs4.element import ResultSet  # type: ignore
+from dateutil import parser
+
+from circuit_maintenance_parser.output import Impact
+from circuit_maintenance_parser.parser import Html, EmailSubjectParser, Status
+
+
+class HtmlParserEquinix(Html):
+    """Custom Parser for HTML portion of Equinix circuit maintenance notifications."""
+
+    def parse_html(self, soup: ResultSet) -> List[Dict]:
+        """Parse an equinix circuit maintenance email.
+
+        Args:
+            soup (ResultSet): beautiful soup object containing the html portion of an email.
+
+        Returns:
+            Dict: The data dict containing circuit maintenance data.
+        """
+        data: Dict[str, Any] = {"circuits": list()}
+
+        impact = self._parse_b(soup.find_all("b"), data)
+        self._parse_table(soup.find_all("th"), data, impact)
+        return [data]
+
+    @staticmethod
+    def _isascii(string):
+        """Python 3.6 compatible way to determine if string is only english characters.
+
+        Args:
+            string (str): string to test if only ascii chars.
+
+        Returns:
+            bool: Returns True if string is ascii only, returns false if the string contains extended unicode characters.
+        """
+        try:
+            string.encode("ascii")
+            return True
+        except UnicodeEncodeError:
+            return False
+
+    def _parse_b(self, b_elements, data):
+        """Parse the <b> elements from the notification to capture start and end times, description, and impact.
+
+        Args:
+            b_elements (): resulting soup object with all <b> elements
+            data (Dict): data from the circuit maintenance
+
+        Returns:
+            impact (Status object): impact of the maintenance notification (used in the parse table function to assign an impact for each circuit).
+        """
+        for b_elem in b_elements:
+            if "UTC:" in b_elem:
+                raw_time = b_elem.next_sibling
+                # for non english equinix notifications
+                # english section is usually at the bottom
+                # this skips the non english line at the top
+                if not self._isascii(raw_time):
+                    continue
```

Comment:
This feels pretty fragile to me but we can go with it for now and fix later if it proves to be an issue.

Reply:
Agreed. Unfortunately seemed to be the best way to detect the non-english characters which don't encode to ascii. If they stop including English, this would break, but than we'd have a larger problem of localization.